message.go

```diff
@@ -139,24 +138,18 @@ func (m *Message) WriteTo(w io.Writer) (int64, error) {
 	return total, nil
 }
 
-// DecodeMessage deseralizes data (as []byte) and creates a new Message
+// DecodeMessage deserializes data (as []byte) and creates a new Message
 func DecodeMessage(b []byte) (*Message, error) {
 	var msg Message
 
-	msg.Timestamp = int64(binary.BigEndian.Uint64(b[:8]))
-	msg.Attempts = binary.BigEndian.Uint16(b[8:10])
-
-	buf := bytes.NewBuffer(b[10:])
-
-	_, err := io.ReadFull(buf, msg.ID[:])
-	if err != nil {
-		return nil, err
+	if len(b) < 11+MsgIDLength {
```

Comment:
i would say no. because the lines below make it clear. it has hardcoded slice ranges from 0-7, 8 to 9, etc. pulling out the number 11 (or 11+MsgIDLengith) and defining it elsewhere feels weird because than you have a bunch of magic numbers except for one that is defined elsewhere. if we go this route, than we should also constant-ize all the other magic numbers IMHO. (which i wouldn't do, seems overkill. what we could do, is document the message format somewhere - if not already exists - and link to it)